src/object.c

```diff
@@ -166,13 +166,12 @@ int git_object_lookup_prefix(
 			error = git_odb_read(&odb_obj, odb, id);
 		}
 	} else {
-		git_oid short_oid;
+		git_oid short_oid = {{ 0 }};
 
 		/* We copy the first len*4 bits from id and fill the remaining with 0s */
```

Comment:
@ethomson: Just took a look at `git_oid__cpy_prefix`. Shouldn't it first memset `out` to all-zero? Currently we just copy the prefix and only set `out->id[len/2] &= 0xF0` without touching the trailing bits. I see other invocations just initialize the oid with `= {{0}}`, but I'd at least like to document the requirement.